garage/tf/policies/base2.py

```diff
@@ -88,6 +95,22 @@ def terminate(self):
         """Clean up operation."""
         pass
 
+    def get_trainable_vars(self):
+        """Get trainable vars."""
+        return self._variable_scope.trainable_variables()
+
+    def get_global_vars(self):
+        """Get global vars."""
+        return self._variable_scope.global_variables()
+
+    def get_regularizable_vars(self):
+        """Get regularizable vars."""
+        reg_vars = [
+            var for var in self.get_trainable_vars()
+            if 'W' in var.name and 'output' not in var.name
```

Comment:
is this a TensorFlow convention?

Reply:
i.e. what makes `'W'` and `'output'` special?

Reply:
It existed in garage before. I will figure that out.

Reply:
now is a good time. it seems like it might be specific to MLPs?

Reply:
Yes. I don't think it is the naming convention of TensirFlow. It could be a convention in garage layers.

Reply:
Yes. I don't think it is the naming convention of TensirFlow. It could be a convention in garage layers.

Reply:
I think in TensorFlow the convention is 'kernel' and 'bias'. 
My thought on this function is, can we add regularizer to the primitives rather than here? I think tf.keras.layer support add_kernel_regularizer.

Reply:
My thought is if we only care about the regularization loss, can we just get that from `tf.losses.get_regularization_losses` so we don't have to keep track of the regularizable variables? It returns a list of regularization losses as Tensor. (https://www.tensorflow.org/api_docs/python/tf/losses/get_regularization_losses)

Reply:
Good idea. I don't have a preference because they serve the same idea. But please test what you decide to use returns the same loss in ddpg.

Reply:
@CatherineSue I agree. I think we can do `tf.layers.dense(x, kernel_regularizer=tf.contrib.layers.l2_regularizer(0.001))` followed by `loss = ordinary_loss + tf.losses.get_regularization_loss()`. Can I do that in another PR?

Reply:
How do you choose l1 or l2 in get_regularization_loss?

Reply:
We pass the regularizer to the dense layer, e.g. `tf.contrib.layers.l2_regularizer(0.001)` and when we get the loss, it will be l2.

Reply:
I see. I think you misunderstand my proposa. By add_regularizer I mean the 'kernel_regularizer' in the argument.

Reply:
Oh I see. By that we will use the same regularizer within a primitive?